core/metrics-core-service/src/main/java/org/hawkular/metrics/core/dropwizard/MetricsInitializer.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.dropwizard;
+
+import static org.hawkular.metrics.model.MetricType.COUNTER;
+import static org.hawkular.metrics.model.MetricType.GAUGE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.hawkular.metrics.core.service.MetricsService;
+import org.hawkular.metrics.model.Metric;
+import org.hawkular.metrics.model.MetricId;
+import org.jboss.logging.Logger;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.ImmutableMap;
+
+import rx.Completable;
+import rx.Observable;
+
+/**
+ * Creates the metric definitions for internal metrics. By internal I mean those metrics that hawkular-metrics collects
+ * about itself. Each metric has two tags - hostname and baseMetric. The hostname defaults to the hostname of the
+ * local host address. The baseMetric tags is for grouping constituent metrics belonging to a DropWizard metric.
+ * Hawkular Metrics does not have a meter metric type for example; so, each of the rates stores in a meter are persisted
+ * as separate gauge metrics.
+ *
+ * @author jsanda
+ */
+public class MetricsInitializer {
+
+    private static Logger logger = Logger.getLogger(MetricsInitializer.class);
+
+    private MetricsService metricsService;
+
+    private MetricRegistry metricRegistry;
+
+    private MetricNameService metricNameService;
+
+    public MetricsInitializer(MetricRegistry metricRegistry, MetricsService metricsService,
+            MetricNameService metricNameService) {
+        this.metricRegistry = metricRegistry;
+        this.metricsService = metricsService;
+        this.metricNameService = metricNameService;
+    }
+
+    public void run() {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
+        logger.info("Creating metrics");
+
+        List<Metric<Double>> gaugeMetrics = new ArrayList<>();
+        List<Metric<Long>> counterMetrics = new ArrayList<>();
+
+        metricRegistry.getMeters(metricNameService).entrySet().forEach(entry -> {
+            String tenantId = MetricNameService.TENANT_ID;
+            String metricName = metricNameService.getMetricName(entry.getKey());
+            Map<String, String> tags = getTags(metricName);
+
+            gaugeMetrics.add(new Metric<>(new MetricId<>(tenantId, GAUGE, metricName + "-1min"), tags));
```

Comment:
I think we should adopt the same naming convention than with the hawkular-reporter, see here: https://github.com/hawkular/hawkular-dropwizard-reporter/blob/master/hawkular-dropwizard-reporter/src/main/java/org/hawkular/client/dropwizard/HawkularReporter.java#L171

So, it should be changed either here or on the reporter side, no? As I understand, parts of a metric name are quite often (but maybe not conventionally) separated with dots.